File: S4-Clients/Java-client/src/main/java/com/ontotext/s4/catalog/ServiceDescriptor.java
/*
 * S4 Java client library
 * Copyright (c) 2014, Ontotext AD, All rights reserved.
 *
 * This library is free software; you can redistribute it and/or
 * modify it under the terms of the GNU Lesser General Public
 * License as published by the Free Software Foundation; either
 * version 3.0 of the License, or (at your option) any later version.
 *
 * This library is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 * Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public
 * License along with this library.
 */
package com.ontotext.s4.catalog;


/**
 * "Struct" class to represent a single service.
 */
public class ServiceDescriptor  {

  /**
   * The service name.
   */
  public String name;

  /**
   * A short fragment of HTML describing the service.
   */
  public String shortDescription;

 /**
   * URL to process documents using the service endpoint.
   */
  public String onlineUrl;

}

File: S4-Clients/C#-client/S4ClientApi/Catalog/ServiceDescriptor.cs
/*
 * S4 C# client library
 * Copyright 2016 Ontotext AD
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Ontotext.S4.catalog
{
    /// <summary>
    /// A model class representing an S4 service
    /// </summary>
    public class ServiceDescriptor
    {
        private String name;

        private String description;

        private String serviceUrl;

        /// <summary>
        /// The service name.
        /// </summary>
        public String Name
        {
            get { return name; }
            set { name = value; }
        }

        /// <summary>
        /// A description of the service.
        /// </summary>
        public String Description
        {
            get { return description; }
            set { description = value; }
        }

        /// <summary>
        ///  URL to process documents using the service endpoint.
        /// </summary>
        public String ServiceUrl
        {
            get { return serviceUrl; }
            set { serviceUrl = value; }
        }
    }
}
